subworkflows/nf-core/fastq_create_umi_consensus_fgbio/main.nf

```diff
@@ -0,0 +1,204 @@
+        BAMFILTER ( dummy_bam_bai, [], [] )
+        ch_versions = ch_versions.mix(BAMFILTER.out.versions)
+        groupready_bam = BAMFILTER.out.bam
+        // deleting dummy file
+        dummyIndexFile.delete()
+
+    } else {
+        // for Adjacency strategy the above is not necessary
+        groupready_bam = ZIPPERBAMS_PRE.out.bam
+    }
+
+    // appropriately tagged reads are now grouped by UMI information
+    // note that in tests ext.args has been set to recommended --edits 1
+    // if UMIs are significantly longer (e.g. 20bp) or have more errors, --edits can be increased
+    // explicit parameter strategy is Adjacency
+    // For multiplex PCR and similar data where reads' genomic positions are fixed by the primers
+    // it is recommended to use --strategy Identity to reduce runtime at the expense of lower accuracy
+    // For duplex UMIs reads MUST be grouped using --strategy paired
+    GROUPREADSBYUMI ( groupready_bam, groupreadsbyumi_strategy )
+    ch_versions = ch_versions.mix(GROUPREADSBYUMI.out.versions)
+
+    // prepare output channel independently on UMI structure
+    consensus_bam = Channel.empty()
+
+    if (duplex){
+        // this is executed if the library contains duplex UMIs
+        CALLDUPLEXCONSENSUS ( GROUPREADSBYUMI.out.bam )
+        ch_versions = ch_versions.mix(CALLDUPLEXCONSENSUS.out.versions)
+        consensus_bam =  CALLDUPLEXCONSENSUS.out.bam
+
+    } else {
+        // using the above created groups, a consensus across reads in the same group
+        // can be called
+        // this will emit a consensus BAM file
+        CALLUMICONSENSUS ( GROUPREADSBYUMI.out.bam )
+        ch_versions = ch_versions.mix(CALLUMICONSENSUS.out.versions)
+        consensus_bam =  CALLUMICONSENSUS.out.bam
+    }
+
+    // please note in FILTERCONSENSUSREADS:
+    // --min-reads is a required argument with no default
+    // --min-base-quality is a required argument with no default
+    // make sure they are specified via ext.args in your config
+    FILTERCONSENSUS ( consensus_bam, fasta )
+    ch_versions = ch_versions.mix(FILTERCONSENSUS.out.versions)
+
+    // now the consensus uBAM needs to be converted into FASTQ again
+    // to be aligned
+    BAM2FASTQ_POST ( FILTERCONSENSUS.out.bam, false )
+    ch_versions = ch_versions.mix(BAM2FASTQ_POST.out.versions)
+
+    aligned_bam_post = Channel.empty()
+
+    if (aligner == "bwa-mem") {
+        // index made available through previous steps
+        BWAMEM1_MEM_POST ( BAM2FASTQ_POST.out.fastq, bwaindex, false )
+        ch_versions = ch_versions.mix(BWAMEM1_MEM_POST.out.versions)
+        aligned_bam_post = BWAMEM1_MEM_POST.out.bam
+    } else {
+        // index made available through previous steps
+        BWAMEM2_MEM_POST ( BAM2FASTQ_POST.out.fastq, bwaindex, false )
+        ch_versions = ch_versions.mix(BWAMEM2_MEM_POST.out.versions)
+        aligned_bam_post = BWAMEM2_MEM.out.bam
+    }
```

Comment:
I guess was an original plan that changed. I must confess I'd prefer to use mix as this could support versions where the aligner is specified by sample? 
I also usually declare the channels first, but if it doesn't throw an error I don't have any particular preference at this stage.